tests/unit/backend/wmg/api/test_expression_rollup.py

```diff
@@ -54,14 +56,326 @@ def test__cell_types_in_same_lineage_are_colinear(self):
+                ["ENSG00000085265", "UBERON:0000955", "CL:0000644", 1, 1, 70, 1000],
+                ["ENSG00000085265", "UBERON:0000955", "CL:0002605", 1, 1, 80, 1000],
+                ["ENSG00000169429", "UBERON:0000955", "CL:0002627", 1, 1, 90, 1000],
+                ["ENSG00000085265", "UBERON:0002113", "CL:0000644", 1, 1, 70, 1000],
+                ["ENSG00000085265", "UBERON:0002113", "CL:0002605", 1, 1, 80, 1000],
+                ["ENSG00000169429", "UBERON:0002113", "CL:0002627", 1, 1, 90, 1000],
+            ],
+            "expected_rolled_up_gene_expression": [
+                ["ENSG00000085265", "UBERON:0000955", "CL:0000127", 2, 2, 150, 1000],
+                ["ENSG00000169429", "UBERON:0000955", "CL:0000127", 1, 1, 90, 1000],
+                ["ENSG00000085265", "UBERON:0000955", "CL:0000644", 1, 1, 70, 1000],
+                ["ENSG00000085265", "UBERON:0000955", "CL:0002605", 1, 1, 80, 1000],
+                ["ENSG00000169429", "UBERON:0000955", "CL:0002627", 1, 1, 90, 1000],
+                ["ENSG00000085265", "UBERON:0002113", "CL:0000127", 2, 2, 150, 1000],
+                ["ENSG00000169429", "UBERON:0002113", "CL:0000127", 1, 1, 90, 1000],
+                ["ENSG00000085265", "UBERON:0002113", "CL:0000644", 1, 1, 70, 1000],
+                ["ENSG00000085265", "UBERON:0002113", "CL:0002605", 1, 1, 80, 1000],
+                ["ENSG00000169429", "UBERON:0002113", "CL:0002627", 1, 1, 90, 1000],
+            ],
+        }
+
+        test_2 = {
+            "name": "no_compare_dim_one_ancestor_cell_type_missing_in_one_tissue_but_exists_in_all_others",
+            # Tissue: "UBERON:0000955" MISSING cell type: "CL:0000127" in input cell counts
+            "input_cell_counts": [
+                ["UBERON:0000955", "CL:0000644", 70],
+                ["UBERON:0000955", "CL:0002605", 80],
+                ["UBERON:0000955", "CL:0002627", 90],
+                ["UBERON:0002113", "CL:0000127", 300],
+                ["UBERON:0002113", "CL:0000644", 70],
+                ["UBERON:0002113", "CL:0002605", 80],
+                ["UBERON:0002113", "CL:0002627", 90],
+            ],
+            # cell count for cell type: "CL:0000127" in Tissue: "UBERON:0000955" GETS AGGREGATED
+            # count because Tissue "UBERON:0000955" CONTAINS cell counts for
+            # descendants of cell type: "CL:0000127"
+            "expected_rolled_up_cell_counts": [
+                ["UBERON:0000955", "CL:0000127", 240],
+                ["UBERON:0000955", "CL:0000644", 70],
+                ["UBERON:0000955", "CL:0002605", 80],
+                ["UBERON:0000955", "CL:0002627", 90],
+                ["UBERON:0002113", "CL:0000127", 540],
+                ["UBERON:0002113", "CL:0000644", 70],
+                ["UBERON:0002113", "CL:0002605", 80],
+                ["UBERON:0002113", "CL:0002627", 90],
+            ],
+            "input_gene_expression": [
+                ["ENSG00000085265", "UBERON:0000955", "CL:0000644", 1, 1, 70, 1000],
```

Comment:
Let's say that we augment the input gene expression DF with a new numeric column that will be rolled up. In that case, these tests need not be updated since the rollup code is mostly agnostic to which numeric columns are being rolled up.

Is that correct? if not, I'm always hesitant to hardcode all the expected inputs / outputs (even though it is the most transparent) if future updates will require these test fixtures to be manually updated.

Reply:
@atarashansky - No these test fixtures need not be updated _if you don't want to check that the new column is rolled up_. Note that I am checking that `sum` and `nnz` are rolled up but you might very well deem that that is enough and and additional columns need not be checked and therefore the test fixture need not change.

However, note that if the `rollup` implementation itself changes where you specifically access a particular column by name (Let's say the new column is called `abc`), then `abc` is not in the test fixture and it would pass a dataframe into `rollup` that doesn't contain the column `abc` and therefore the `rollup` function will throw an exception - This is the expected behavior and demonstrates the value of the unit test.